components/support/base/src/main/java/mozilla/components/support/base/observer/ObserverRegistry.kt

```diff
@@ -95,6 +99,8 @@ class ObserverRegistry<T> : Observable<T> {
             }
             observers.clear()
             pausedObservers.clear()
+            lifecycleObservers.clear()
+            viewObservers.clear()
```

Comment:
It looks like there's a second bug hiding here: First we go through all observers and remove a lifecycle observer if it exists, but we never do that for an potentially existing view observer. I'll file a follow-up issue for that. Maybe we should play safe and just call `unregister()` for every observer here. Anyhow, different issue!

Reply:
Follow-up: https://github.com/mozilla-mobile/android-components/issues/3278